### PR TITLE
Mods

### PR DESCRIPTION
**Pull request by siebrenf**
**What problem is the PR solving / What's new?**
making some functions more modular (for integration with ANANSE again)

oh, and I updated the check for the minimal snakemake version, since snakemake now checks for `snakefile_main`.

**Checklist**
- [x] I made a PR to develop (not master)
- [x] These changes are covered by the tests

### CODE REVIEW

File: seq2science/rules/configuration_generic.smk [hunk 1]
```diff
@@ -508,7 +474,7 @@ wildcard_constraints:
 
 
 # make sure the snakemake version corresponds to version in environment
-min_version("5.18")
+min_version("6.12.3")
```
Comment (Maarten-vd-Sande) [inline seq2science/rules/configuration_generic.smk:477]: good one!

File: seq2science/rules/qc.smk [hunk 2]
```diff
@@ -751,7 +751,7 @@ rule multiqc_schema:
         order = -954
         deseq2_imgs = ""
         deseq2_order = ""
-        for contrast in expand_contrasts(samples, config):
+        for contrast in expand_contrasts(samples, config["contrasts"]):
```
Comment (Maarten-vd-Sande) [inline seq2science/rules/qc.smk:754]: where is contrasts being set? Do you mean `DE_contrasts`?

File: seq2science/rules/qc.smk [hunk 3]
```diff
@@ -854,10 +854,10 @@ def get_qc_files(wildcards):
         qc["files"].update(files)
 
     # DESeq2 all contrast plots
-    if "get_contrasts" in globals():
+    if "DE_contrasts" in globals():
```
Comment (Maarten-vd-Sande) [inline seq2science/rules/qc.smk:857]: should we maybe make `DE_contrasts` just an empty list as default? Not sure what is better here

Comment (Maarten-vd-Sande) [reply seq2science/rules/qc.smk:857]: Although maybe checking if it exists is the easier solution indeed

**Review by Maarten-vd-Sande — APPROVED** [review]
Nice refactoring, makes it much cleaner!

File: seq2science/rules/qc.smk (hunk 2 above)
Comment (siebrenf) [reply seq2science/rules/qc.smk:754]: `DE_contrasts` contains the file paths to each DESeq2 output.
`expand_contrasts()` returns the actual contrasts, which are parsed from the config.

so this line is correct, however, we could also make it into another global list, or run `config["contrasts"] = expand_contrasts(samples, config["contrasts"])` somewhere in the config. Do you have a preference?

File: seq2science/rules/qc.smk (hunk 3 above)
Comment (siebrenf) [reply seq2science/rules/qc.smk:857]: using a global variable or checking if a variable exists. seems like chosing the lesser evil :p

**Pull request merged**